### DISCUSSION TITLE
how to set hive SQLALCHEMY URI  with username and password?

### DISCUSSION THREAD

**Discussion by zhugw** [apache/superset#18699] [Q&A / Help] [answered]
Version:  1.4.1
Want to connect to hive, python connect hive code as below

```
from pyhive import hive

host_name = "192.168.0.38"
port = 10000
user = "admin"
password = "password"
database="test_db"

def hiveconnection(host_name, port, user,password, database):
    conn = hive.Connection(host=host_name, port=port, username=user, password=password,
                           database=database, auth='CUSTOM')
    cur = conn.cursor()
    cur.execute('select item_sk,reason_sk, account_credit from returns limit 5')
    result = cur.fetchall()

    return result
```

but how to set hive `SQLALCHEMY URI`  with username and password?
```
hive://hive@{hostname}:{port}/{database}
```

**Comment by zhaoyongjie** [comment]
Hi @zhugw, the issue list is used to track bugs now, so I transfer this topic to the discussion board.

**Comment by zhaoyongjie** [comment, marked as answer]
Hi @zhugw, it depends on how to set authorization in Hive. When you are using HiveServer2 and plain authorization, try following URI
```
hive://username:password@hostname:port/database
```

When you are using https + HiveServer2, try following URI:
```
hive+https://username:password@localhost:10000/
```

#### reference
- https://cwiki.apache.org/confluence/display/hive/setting+up+hiveserver2#SettingUpHiveServer2-RunninginHTTPMode
- https://github.com/dropbox/PyHive#sqlalchemy

**Reply by zhugw** [reply]
Thanks! It works.
And in my case should explicitly specify `?auth=CUSTOM`
```
hive://username:password@hostname:port/database?auth=CUSTOM
```

**Reply by ghost** [reply]
thanks !
hive need privilege and can't see database schema .
use this connect way can solve .
